File: week-6/sentimental-readability/readability.py
from cs50 import get_string

# Get input text
text = get_string("Text: ")

# Initialize counters
letters = sum(1 for char in text if char.isalpha())
words = sum(1 for char in text if char.isspace()) + 1
sentences = sum(1 for char in text if char in ".!?")

# Calculate L and S
l = letters / words * 100
s = sentences / words * 100

# Calculate the Coleman-Liau index
index = round(0.0588 * l - 0.296 * s - 15.8)

# Print the grade level
if index >= 16:
    print("Grade 16+")
elif index >= 1:
    print(f"Grade {index}")
else:
    print("Before Grade 1")

File: week-2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int words(string s)
{
    int w = 0;
    int n = strlen(s);
    for (int i = 0; i < n; i++)
        w += (s[i] == ' ');
    return (w + 1);
}

int letters(string s)
{
    int l = 0;
    int n = strlen(s);
    for (int i = 0; i < n; i++)
        l += (isupper(s[i]) || islower(s[i]));
    return l;
}

int sentences(string s)
{
    int l = 0;
    int n = strlen(s);
    for (int i = 0; i < n; i++)
        l += (s[i] == '.' || s[i] == '!' || s[i] == '?');
    return l;
}

int main()
{
    string text = get_string("Text: ");
    int W = words(text);
    int L = letters(text);
    int S = sentences(text);

    int index = round(0.0588 * (L / (float) W * 100.0) - 0.296 * (S / (float) W * 100.0) - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
